pf/internal/schemashim/type_schema.go

```diff
@@ -56,19 +57,36 @@ func (*typeSchema) Required() bool  { return false }
 func (*typeSchema) Sensitive() bool { return false }
 
 func (s *typeSchema) Elem() interface{} {
+	asObjectType := func(typ any) (shim.Resource, bool) {
```

Comment:
This doesn't feel right. typeSchema is in the represents a Schema that's a simple type, not block or attribute, and it's testing it for types.ObjectType on line 69, and recurring into itself. Let me check this PR out and play with this a bit.